src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -0,0 +1,607 @@
+                    declare_index->index_columns->children.begin(), declare_index->index_columns->children.end());
+            if (startsWith(declare_index->index_type, "PRIMARY_KEY_"))
+                primary_keys->arguments->children.insert(keys->arguments->children.end(),
+                    declare_index->index_columns->children.begin(), declare_index->index_columns->children.end());
+        }
+    }
+
+    for (const auto & declare_column_ast : columns_define->children)
+    {
+        const auto & declare_column = declare_column_ast->as<MySQLParser::ASTDeclareColumn>();
+
+        if (declare_column->column_options)
+        {
+            if (const auto * options = declare_column->column_options->as<MySQLParser::ASTDeclareOptions>())
+            {
+                if (options->changes.count("unique_key"))
+                    unique_keys->arguments->children.emplace_back(std::make_shared<ASTIdentifier>(declare_column->name));
+
+                if (options->changes.count("primary_key"))
+                    primary_keys->arguments->children.emplace_back(std::make_shared<ASTIdentifier>(declare_column->name));
+
+                if (options->changes.count("auto_increment"))
+                    increment_columns.emplace(declare_column->name);
+            }
+        }
+    }
+
+    return std::make_tuple(
+        getNames(*primary_keys, context, columns), getNames(*unique_keys, context, columns), getNames(*keys, context, columns), increment_columns);
+}
+
+static String getUniqueColumnName(NamesAndTypesList columns_name_and_type, const String & prefix)
+{
+    const auto & is_unique = [&](const String & column_name)
+    {
+        for (const auto & column_name_and_type : columns_name_and_type)
+        {
+            if (column_name_and_type.name == column_name)
+                return false;
+        }
+
+        return true;
+    };
+
+    if (is_unique(prefix))
+        return prefix;
+
+    for (size_t index = 0; ; ++index)
+    {
+        const String & cur_name = prefix + "_" + toString(index);
+        if (is_unique(cur_name))
+            return cur_name;
+    }
+}
+
+static ASTPtr getPartitionPolicy(const NamesAndTypesList & primary_keys)
+{
+    const auto & numbers_partition = [&](const String & column_name, const DataTypePtr & type, size_t type_max_size)
+    {
+        ASTPtr column = std::make_shared<ASTIdentifier>(column_name);
+
+        if (type->isNullable())
+            column = makeASTFunction("assumeNotNull", column);
+
+        return makeASTFunction("intDiv", column, std::make_shared<ASTLiteral>(UInt64(type_max_size / 1000)));
+    };
+
+    ASTPtr best_partition;
+    size_t best_size = 0;
+    for (const auto & primary_key : primary_keys)
+    {
+        DataTypePtr type = primary_key.type;
+        WhichDataType which(type);
+
+        if (which.isNullable())
+        {
+            type = (static_cast<const DataTypeNullable &>(*type)).getNestedType();
```

Comment:
Since `type` is replaced with nested type, `numbers_partition` will not add `assumeNotNull` for `Nullable` column.

Reply:
Done.